pandas/core/arrays/categorical.py

```diff
@@ -2305,6 +2306,25 @@ def replace(self, to_replace, value, inplace: bool = False):
         if not inplace:
             return cat
 
+    # ------------------------------------------------------------------------
+    # String methods interface
+    def _str_map(self, f, na_value=np.nan, dtype=np.dtype(object)):
+        # Optimization to apply the callable `f` to the categories once
+        # and rebuild the result by `take`ing from the result with the codes.
+        # Returns the same type as the object-dtype implementation though.
+        from pandas.core.arrays import PandasArray
+
+        categories = self.categories
+        codes = self.codes
+        result = PandasArray(categories.to_numpy())._str_map(f, na_value, dtype)
+        return take_1d(result, codes, fill_value=na_value)
+
+    def _str_get_dummies(self, sep="|"):
+        # sep may not be in categories. Just bail on this.
```

Comment:
does this mean "we are assuming sep is not in self.categories", or "sep might not be in self.categories"?

Reply:
sep might not be in self.categories.